index.js

```diff
@@ -48,51 +48,255 @@ const player = {
+  player.songs.push({ title, album, artist, duration, id })
+  return id
 }
 
 function removePlaylist(id) {
-  // your code here
+  if (!checkId(player.playlists, id)) throw new Error('ID is not found')
+  let correctPlaylist = findPlaylistById(id)
+  player.playlists.splice(correctPlaylist, 1)
 }
 
-function createPlaylist(name, id) {
-  // your code here
+function createPlaylist(name, id = generateNewId(player.playlists)) {
+  if (checkId(player.playlists, id))
+    throw new Error('ID already exist, change the ID or omit it')
+  player.playlists.push({ name, id, songs: [] })
+  return id
 }
 
 function playPlaylist(id) {
-  // your code here
+  if (!checkId(player.playlists, id))
+    throw new Error("ID isn't exist, change the ID")
+  let correctPlaylist = findPlaylistById(id)
+  for (let j = 0; j < correctPlaylist.songs.length; j++) {
+    //run on the songs array inside the wanted playlist
+    playSong(correctPlaylist.songs[j])
+  }
+  return id
 }
 
 function editPlaylist(playlistId, songId) {
-  // your code here
+  if (!checkId(player.songs, songId))
+    throw new Error("ID isn't exist, change the ID")
+  if (!checkId(player.playlists, playlistId))
+    throw new Error("ID isn't exist, change the ID")
+  let correctPlaylist = findPlaylistById(playlistId)
+  for (let j = 0; j < correctPlaylist.songs.length; j++) {
+    //runs on the songs array in the playlist
+    if (songId === correctPlaylist.songs[j]) {
+      //if the song ID exists in the playlist
+      correctPlaylist.songs.splice(j, 1) //removes it
+    } else {
+      correctPlaylist.songs.push(songId)
+    }
+    if (correctPlaylist.songs.length === 0) {
+      //if it was the only song in the playlist
+      removePlaylist(correctPlaylist.id) //remove this playlist
+    }
+  }
 }
 
 function playlistDuration(id) {
-  // your code here
+  let correctPlaylist = findPlaylistById(id) //correctPlaylist contain the wanted playlist
+  let saveSongId = 0,
+    sum = 0
+  for (let i = 0; i < correctPlaylist.songs.length; i++) {
+    //run on the songs array inside this playlist
+    saveSongId = correctPlaylist.songs[i]
+    for (let j = 0; j < player.songs.length; j++) {
+      //run on the songs array
+      if (player.songs[j].id === saveSongId) sum += player.songs[j].duration
+    }
+  }
+  return sum
 }
 
 function searchByQuery(query) {
-  // your code here
+  let tempQuery = query.toUpperCase()
+  const results = { songs: [], playlists: [] }
+  for (let i = 0; i < player.playlists.length; i++) {
+    //for playlists
+    if (player.playlists[i].name.toUpperCase().includes(tempQuery)) {
+      results.playlists.push(player.playlists[i])
+      results.playlists.sort((a, b) => {
+        if (a.name.toUpperCase() < b.name.toUpperCase()) return -1
+      })
+    }
+  }
+  for (let i = 0; i < player.songs.length; i++) {
+    //for songs
+    if (
+      player.songs[i].album.toUpperCase().includes(tempQuery) ||
```

Comment:
move to a function or put all the fields you want to check to array and loop through them